### PR TITLE
[FIX] mail: broken template

### PR DESCRIPTION
**Pull request by anko-odoo**
If sale order confirmation template was broken in some way eg. field no longer exists in the model. When mail confirmation was send it caused an error and additionaly, Sale Order was not confirmed even when transaction was done, which could result in customers paying multiple times for the same Sale Order.

Now, in case of faulty template, email still will be send, but without the template body.

opw-3957553

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/wizard/mail_compose_message.py [hunk 1]
```diff
@@ -231,7 +233,11 @@ def _compute_body(self):
         template, reset it. """
         for composer in self:
             if composer.template_id:
-                composer._set_value_from_template('body_html', 'body')
+                try:
+                    composer._set_value_from_template('body_html', 'body')
```
Comment (reth-odoo) [inline addons/mail/wizard/mail_compose_message.py:237]: So it's an issue with qweb? Wouldn't it make more sense to wrap the actual _generate_template call if fixing it here?

**Review by reth-odoo — COMMENTED** [review]
If the email blocks an automated flow, shouldn't the error be caught in the actual flow?

When using the composer normally you probably want to know the template is broken rather than get no body at all

**Comment by reth-odoo** [comment]
@anko-odoo pushed my suggestion on here, the reason you had issues was probably due to the exception happening inside a compute so the ORM was trying to recompute it at the end and failing again. Using a savepoint around the mail stuff will fix that and should be fine as only mail stuff should happen in there...

Remains to write a test, ideally.

**Comment by reth-odoo** [comment]
@anko-odoo :wave: just reminder to squash / write test if you're not busy on other things
Closing this tab so feel free to ping again once it's done :)

Cheers

**Comment by anko-odoo** [comment]
@reth-odoo After further investigation and discussion with @Feyensv I think the current direction of the solution is not the correct one as 1. As stated in one of the guru meetings, savepoints should be avoided. 2. The current solution will confirm the Sale Order but no confirmation would be sent, possibly leading to many confused clients. Also, this is not only sale confirmation problem, but rather all template problem.

**Comment by reth-odoo** [comment]
To avoid the savepoint you can check on write and prevent writing invalid templates as suggested (I couldn't find a task on our side).

For the template not being sent, yes but I would say not confirming a paid sale order for example is worse than not receiving an email. Don't really see what generic fix we could provide for "we can't send the email" that seems like a case by case kind of issue.

**Pull request closed without merging**